File: Programmers/연습문제/Level 1/예산/Solution.js
function solution(d, budget) {
    var d = d.sort((a, b) => (a-b));
    while (sum(d) > budget) {
        d.pop();
    }
     const answer = getAnswer(d);
     return answer;
}

function sort(arr) {
    return arr.sort((a, b) => (a-b));
}

function sum(arr) {
    let result = 0;
    for (const num of arr) {
        result += num;
    }
    return result;
}

function getAnswer(arr) {
    return arr.length;
}


// TC
const d = [1,3,2,5,4];
const budget = 9;
console.log(solution(d, budget));

File: Programmers/연습문제/Level 1/예산/Solution.swift
import Foundation

func solution(_ d:[Int], _ budget:Int) -> Int {
    var total = 0
    var count = 0
    d.sorted().map {
        if total + $0 <= budget {
            total += $0
            count += 1
        }
    }
    return count
}

    
// TC
let d = [1, 3, 2, 5, 4]
let budget = 9
print(solution(d, budget)) // 3
